src/WebView.js

```diff
@@ -20,6 +25,38 @@ class WebView extends WebKit.WebView {
     );
     user_content_manager.add_style_sheet(stylesheet);
   }
+
+  #onDecidePolicy = (_self, decision, decision_type) => {
+    console.debug(
+      "decide-policy",
+      getEnum(WebKit.PolicyDecisionType, decision_type),
+    );
+
+    if (decision_type === WebKit.PolicyDecisionType.NAVIGATION_ACTION) {
+      const navigation_action = decision.get_navigation_action();
+      const url = navigation_action.get_request().get_uri();
+      console.debug(
+        "navigation",
+        getEnum(WebKit.NavigationType, navigation_action.get_navigation_type()),
+        url,
+      );
+
+      const scheme = GLib.Uri.peek_scheme(url);
+      if (scheme !== "file") {
+        decision.ignore();
+        Gtk.show_uri(this.get_root(), url, Gdk.CURRENT_TIME);
```

Comment:
I think ``show_uri`` is deprecated, shall we use ``Gtk.UriLauncher.launch`` instead?